contracts/SportMarkets/LiquidityPool/AMMLiquidityPool.sol

```diff
@@ -0,0 +1,497 @@
+    /* ========== LIBRARIES ========== */
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    struct InitParams {
+        address _owner;
+        ISportsAMM _sportsAmm;
+        IERC20Upgradeable _sUSD;
+        uint _roundLength;
+        uint _maxAllowedDeposit;
+        uint _minDepositAmount;
+        uint _maxAllowedUsers;
+    }
+
+    /* ========== CONSTANTS ========== */
+    uint private constant HUNDRED = 1e20;
+    uint private constant ONE = 1e18;
+
+    /* ========== STATE VARIABLES ========== */
+
+    ISportsAMM public sportsAMM;
+    IERC20Upgradeable public sUSD;
+
+    bool public started;
+
+    uint public round;
+    uint public roundLength;
+    uint public firstRoundStartTime;
+
+    mapping(uint => address) public roundPools;
+
+    mapping(uint => address[]) public usersPerRound;
+    mapping(uint => mapping(address => bool)) public userInRound;
+
+    mapping(uint => mapping(address => uint)) public balancesPerRound;
+    mapping(uint => uint) public allocationPerRound;
+
+    mapping(address => bool) public withdrawalRequested;
+
+    mapping(uint => address[]) public tradingMarketsPerRound;
+    mapping(uint => mapping(address => bool)) public isTradingMarketInARound;
+
+    mapping(uint => uint) public profitAndLossPerRound;
+    mapping(uint => uint) public cumulativeProfitAndLoss;
+
+    uint public maxAllowedDeposit;
+    uint public minDepositAmount;
+    uint public maxAllowedUsers;
+    uint public usersCurrentlyInPool;
+
+    address public defaultLiquidityProvider;
+
+    /// @return The address of the Staking contract
+    IStakingThales public stakingThales;
+
+    uint public stakedThalesMultiplier;
+
+    address public poolRoundMastercopy;
+
+    mapping(address => bool) public whitelistedDeposits;
+
+    uint public totalDeposited;
+
+    /* ========== CONSTRUCTOR ========== */
+
+    function initialize(InitParams calldata params) external initializer {
+        setOwner(params._owner);
+        initNonReentrant();
+        sportsAMM = ISportsAMM(params._sportsAmm);
+
+        sUSD = params._sUSD;
+        roundLength = params._roundLength;
+        maxAllowedDeposit = params._maxAllowedDeposit;
+        minDepositAmount = params._minDepositAmount;
+        maxAllowedUsers = params._maxAllowedUsers;
+
+        sUSD.approve(address(sportsAMM), type(uint256).max);
+    }
+
+    /// @notice Start pool and begin round #1
+    function start() external onlyOwner {
+        require(!started, "Liquidity pool has already started");
+        require(allocationPerRound[1] > 0, "can not start with 0 deposits");
+        round = 1;
+        firstRoundStartTime = block.timestamp;
+        started = true;
+    }
+
+    /// @notice Deposit funds from user into pool for the next round
+    /// @param amount Value to be deposited
+    function deposit(uint amount) external canDeposit(amount) nonReentrant whenNotPaused {
```

Comment:
although is cleaner, modifier used only once might produce higher contract size

Reply:
Its a reused approach from the vaults code